crates/next-core/src/next_client_reference/ecmascript_client_reference/ecmascript_client_reference_transition.rs

```diff
@@ -47,46 +47,57 @@ impl Transition for NextEcmascriptClientReferenceTransition {
         self: Vc<Self>,
         source: Vc<Box<dyn Source>>,
         module_asset_context: Vc<ModuleAssetContext>,
-        _reference_type: Value<ReferenceType>,
+        reference_type: Value<ReferenceType>,
     ) -> Result<Vc<ProcessResult>> {
+        let part = match &*reference_type {
+            ReferenceType::EcmaScriptModules(EcmaScriptModulesReferenceSubType::ImportPart(
+                part,
+            )) => Some(*part),
+            _ => None,
+        };
+
         let module_asset_context = self.process_context(module_asset_context);
 
         let this = self.await?;
 
-        let ident = source.ident().await?;
-        let ident_path = ident.path.await?;
+        let ident = match part {
+            Some(part) => source.ident().with_part(part),
+            None => source.ident(),
+        };
+        let ident_ref = ident.await?;
+        let ident_path = ident_ref.path.await?;
         let client_source = if ident_path.path.contains("next/dist/esm/") {
-            let path = ident.path.root().join(
+            let path = ident_ref.path.root().join(
                 ident_path
                     .path
                     .replace("next/dist/esm/", "next/dist/")
                     .into(),
             );
-            Vc::upcast(FileSource::new_with_query(path, ident.query))
+            Vc::upcast(FileSource::new_with_query(path, ident_ref.query))
         } else {
             source
         };
-        let client_module = this
-            .client_transition
-            .process(
-                client_source,
-                module_asset_context,
-                Value::new(ReferenceType::Entry(
-                    EntryReferenceSubType::AppClientComponent,
-                )),
-            )
-            .module();
+        let client_module = this.client_transition.process(
+            client_source,
+            module_asset_context,
+            Value::new(ReferenceType::Entry(
```

Comment:
That won't work if you don't pass `part` here, does it? Otherwise is just references the full module